source/common/config/delta_subscription_state.cc

```diff
@@ -202,6 +317,39 @@ DeltaSubscriptionState::getNextRequestAckless() {
   return request;
 }
 
+bool DeltaSubscriptionState::isInitialRequestForLegacyWildcard() {
+  if (in_initial_legacy_wildcard_) {
+    requested_resource_state_.insert_or_assign(Wildcard, ResourceState::waitingForServer());
+    ASSERT(requested_resource_state_.contains(Wildcard));
+    ASSERT(!wildcard_resource_state_.contains(Wildcard));
+    ASSERT(!ambiguous_resource_state_.contains(Wildcard));
+    return true;
+  }
+
+  // If we are here, this means that we lost our initial wildcard mode, because we subscribed to
+  // something in the past. We could still be in the situation now that all we are subscribed to now
+  // is wildcard resource, so in such case try to send a legacy wildcard subscription request
```

Comment:
This is only material when we reconnect after a dropped stream? Or is there some other situation it is useful to recover the legacy wildcard behavior? Just trying to read this with beginners mind (it's been a while since I've reviewed).

Reply:
> This is only material when we reconnect after a dropped stream?

Yes.

> Or is there some other situation it is useful to recover the legacy wildcard behavior? Just trying to read this with beginners mind (it's been a while since I've reviewed).

Not really. This function is invoked only when we are about to send the initial request on the stream. And there is no way to recover the legacy wildcard behavior on a stream after sending the initial request that wasn't empty. We can go back to legacy wildcard behavior only with a new stream.